network/builder/src/builder.rs

```diff
@@ -233,10 +230,23 @@ impl NetworkBuilder {
 
         network_builder.discovery_listeners = Some(Vec::new());
         for discovery_method in config.discovery_methods() {
+            let reconfig_listener = if matches!(discovery_method, DiscoveryMethod::Onchain) {
```

Comment:
I think you can just do `discovery_method == DiscoveryMethod::Onchain`, unless there's some inner data in that one.  We actually have 2 `discovery` enums I think for that reason

Reply:
Done, thanks!